Intermittent test failure: test_analysis_results_view_data_1_Complete

Getting this pretty reliably locally:


```
======================================================================
FAIL: test_analysis_results_view_data_1_Complete (experimenter.visualization.tests.api.test_views.TestVisualizationView)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/usr/local/lib/python3.9/unittest/case.py", line 593, in run
    self._callTestMethod(testMethod)
  File "/usr/local/lib/python3.9/unittest/case.py", line 550, in _callTestMethod
    method()
  File "/usr/local/lib/python3.9/site-packages/parameterized/parameterized.py", line 533, in standalone_func
    return func(*(a + p.args), **p.kwargs)
  File "/usr/local/lib/python3.9/unittest/mock.py", line 1337, in patched
    return func(*newargs, **newkeywargs)
  File "/usr/local/lib/python3.9/site-packages/parameterized/parameterized.py", line 81, in dummy_func
    return orgfunc(*args, **kwargs)
  File "/usr/local/lib/python3.9/unittest/mock.py", line 1337, in patched
    return func(*newargs, **newkeywargs)
  File "/app/experimenter/visualization/tests/api/test_views.py", line 151, in test_analysis_results_view_data
    self.assertEqual(FULL_DATA, json_data)
  File "/usr/local/lib/python3.9/unittest/case.py", line 831, in assertEqual
    assertion_func(first, second, msg=msg)
  File "/usr/local/lib/python3.9/unittest/case.py", line 1130, in assertDictEqual
    self.fail(self._formatMessage(msg, standardMsg))
  File "/usr/local/lib/python3.9/unittest/case.py", line 670, in fail
    raise self.failureException(msg)
AssertionError: {'daily': [{'point': 12, 'upper': 13, 'lowe[7157 chars]alse} != {'show_analysis': False, 'daily': [{'point'[6037 chars]}}}}}
  {'daily': [{'branch': 'control',
              'lower': 10,
              'metric': 'identity',
              'point': 12,
              'statistic': 'count',
              'upper': 13,
              'window_index': '1'},
             {'branch': 'variant',
              'lower': 10,
              'metric': 'identity',
              'point': 12,
              'statistic': 'count',
              'upper': 13,
              'window_index': '1'},
             {'branch': 'variant',
              'lower': 10,
              'metric': 'some_count',
              'point': 12,
              'statistic': 'mean',
              'upper': 13,
              'window_index': '1'},
             {'branch': 'variant',
              'comparison': 'difference',
              'lower': 10,
              'metric': 'search_count',
              'point': 12,
              'statistic': 'mean',
              'upper': 13,
              'window_index': '1'},
             {'branch': 'variant',
              'comparison': 'difference',
              'lower': -5,
              'metric': 'retained',
              'point': -2,
              'statistic': 'binomial',
              'upper': -1,
              'window_index': '1'},
             {'branch': 'control',
              'comparison': 'difference',
              'lower': -5,
              'metric': 'retained',
              'point': 12,
              'statistic': 'binomial',
              'upper': 13,
              'window_index': '1'},
             {'branch': 'control',
              'lower': 2,
              'metric': 'object-based-client-driven-initiative_ever_used',
              'point': 4,
              'statistic': 'binomial',
              'upper': 8,
              'window_index': '1'},
             {'branch': 'variant',
              'lower': 2,
              'metric': 'object-based-client-driven-initiative_ever_used',
              'point': 4,
              'statistic': 'binomial',
              'upper': 8,
              'window_index': '1'},
             {'branch': 'control',
              'lower': 2,
              'metric': 'multi-layered-homogeneous-graphic-interface_ever_used',
              'point': 4,
              'statistic': 'binomial',
              'upper': 8,
              'window_index': '1'},
             {'branch': 'variant',
              'lower': 2,
              'metric': 'multi-layered-homogeneous-graphic-interface_ever_used',
              'point': 4,
              'statistic': 'binomial',
              'upper': 8,
              'window_index': '1'}],
   'other_metrics': {'some_count': 'Some Count'},
   'overall': {'control': {'branch_data': {'identity': {'absolute': {'all': [{'lower': 10,
                                                                              'point': 12,
                                                                              'upper': 13}],
                                                                     'first': {'lower': 10,
                                                                               'point': 12,
                                                                               'upper': 13}},
                                                        'difference': {'all': [],
                                                                       'first': {}},
                                                        'percent': 50,
                                                        'relative_uplift': {'all': [],
                                                                            'first': {}}},
                                           'multi-layered-homogeneous-graphic-interface_ever_used': {'absolute': {'all': [{'count': 48,
                                                                                                                           'lower': 2,
                                                                                                                           'point': 4,
                                                                                                                           'upper': 8}],
                                                                                                                  'first': {'count': 48,
                                                                                                                            'lower': 2,
                                                                                                                            'point': 4,
                                                                                                                            'upper': 8}},
                                                                                                     'difference': {'all': [],
                                                                                                                    'first': {}},
                                                                                                     'relative_uplift': {'all': [],
                                                                                                                         'first': {}}},
-                                          'object-based-client-driven-initiative_ever_used': {'absolute': {'all': [{'count': 48,
-                                                                                                                    'lower': 2,
-                                                                                                                    'point': 4,
-                                                                                                                    'upper': 8}],
-                                                                                                           'first': {'count': 48,
-                                                                                                                     'lower': 2,
-                                                                                                                     'point': 4,
-                                                                                                                     'upper': 8}},
-                                                                                              'difference': {'all': [],
-                                                                                                             'first': {}},
-                                                                                              'relative_uplift': {'all': [],
-                                                                                                                  'first': {}}},
                                           'retained': {'absolute': {'all': [],
                                                                     'first': {}},
                                                        'difference': {'all': [{'lower': -5,
                                                                                'point': 12,
                                                                                'upper': 13},
                                                                               {'lower': -5,
                                                                                'point': 12,
                                                                                'upper': 13}],
                                                                       'first': {'lower': -5,
                                                                                 'point': 12,
                                                                                 'upper': 13}},
                                                        'relative_uplift': {'all': [],
                                                                            'first': {}},
                                                        'significance': 'neutral'}},
                           'is_control': False},
               'variant': {'branch_data': {'identity': {'absolute': {'all': [{'lower': 10,
                                                                              'point': 12,
                                                                              'upper': 13}],
                                                                     'first': {'lower': 10,
                                                                               'point': 12,
                                                                               'upper': 13}},
                                                        'difference': {'all': [],
                                                                       'first': {}},
                                                        'percent': 50,
                                                        'relative_uplift': {'all': [],
                                                                            'first': {}}},
                                           'multi-layered-homogeneous-graphic-interface_ever_used': {'absolute': {'all': [{'count': 48,
                                                                                                                           'lower': 2,
                                                                                                                           'point': 4,
                                                                                                                           'upper': 8}],
                                                                                                                  'first': {'count': 48,
                                                                                                                            'lower': 2,
                                                                                                                            'point': 4,
                                                                                                                            'upper': 8}},
                                                                                                     'difference': {'all': [],
                                                                                                                    'first': {}},
                                                                                                     'relative_uplift': {'all': [],
                                                                                                                         'first': {}}},
-                                          'object-based-client-driven-initiative_ever_used': {'absolute': {'all': [{'count': 48,
-                                                                                                                    'lower': 2,
-                                                                                                                    'point': 4,
-                                                                                                                    'upper': 8}],
-                                                                                                           'first': {'count': 48,
-                                                                                                                     'lower': 2,
-                                                                                                                     'point': 4,
-                                                                                                                     'upper': 8}},
-                                                                                              'difference': {'all': [],
-                                                                                                             'first': {}},
-                                                                                              'relative_uplift': {'all': [],
-                                                                                                                  'first': {}}},
                                           'retained': {'absolute': {'all': [],
                                                                     'first': {}},
                                                        'difference': {'all': [{'lower': -5,
                                                                                'point': -2,
                                                                                'upper': -1},
                                                                               {'lower': -5,
                                                                                'point': -2,
                                                                                'upper': -1}],
                                                                       'first': {'lower': -5,
                                                                                 'point': -2,
                                                                                 'upper': -1}},
                                                        'relative_uplift': {'all': [],
                                                                            'first': {}},
                                                        'significance': 'negative'},
                                           'search_count': {'absolute': {'all': [],
                                                                         'first': {}},
                                                            'difference': {'all': [{'lower': 10,
                                                                                    'point': 12,
                                                                                    'upper': 13}],
                                                                           'first': {'lower': 10,
                                                                                     'point': 12,
                                                                                     'upper': 13}},
                                                            'relative_uplift': {'all': [],
                                                                                'first': {}},
                                                            'significance': 'positive'},
                                           'some_count': {'absolute': {'all': [{'lower': 10,
                                                                                'point': 12,
                                                                                'upper': 13}],
                                                                       'first': {'lower': 10,
                                                                                 'point': 12,
                                                                                 'upper': 13}},
                                                          'difference': {'all': [],
                                                                         'first': {}},
                                                          'relative_uplift': {'all': [],
                                                                              'first': {}}}},
                           'is_control': False}},
   'show_analysis': False,
   'weekly': {'control': {'branch_data': {'identity': {'absolute': {'all': [{'lower': 10,
                                                                             'point': 12,
                                                                             'upper': 13,
                                                                             'window_index': '1'}],
                                                                    'first': {'lower': 10,
                                                                              'point': 12,
                                                                              'upper': 13,
                                                                              'window_index': '1'}},
                                                       'difference': {'all': [],
                                                                      'first': {}},
                                                       'relative_uplift': {'all': [],
                                                                           'first': {}}},
                                          'multi-layered-homogeneous-graphic-interface_ever_used': {'absolute': {'all': [{'lower': 2,
                                                                                                                          'point': 4,
                                                                                                                          'upper': 8,
                                                                                                                          'window_index': '1'}],
                                                                                                                 'first': {'lower': 2,
                                                                                                                           'point': 4,
                                                                                                                           'upper': 8,
                                                                                                                           'window_index': '1'}},
                                                                                                    'difference': {'all': [],
                                                                                                                   'first': {}},
                                                                                                    'relative_uplift': {'all': [],
                                                                                                                        'first': {}}},
-                                         'object-based-client-driven-initiative_ever_used': {'absolute': {'all': [{'lower': 2,
-                                                                                                                   'point': 4,
-                                                                                                                   'upper': 8,
-                                                                                                                   'window_index': '1'}],
-                                                                                                          'first': {'lower': 2,
-                                                                                                                    'point': 4,
-                                                                                                                    'upper': 8,
-                                                                                                                    'window_index': '1'}},
-                                                                                             'difference': {'all': [],
-                                                                                                            'first': {}},
-                                                                                             'relative_uplift': {'all': [],
-                                                                                                                 'first': {}}},
                                          'retained': {'absolute': {'all': [],
                                                                    'first': {}},
                                                       'difference': {'all': [{'lower': -5,
                                                                               'point': 12,
                                                                               'upper': 13,
                                                                               'window_index': '1'}],
                                                                      'first': {'lower': -5,
                                                                                'point': 12,
                                                                                'upper': 13,
                                                                                'window_index': '1'}},
                                                       'relative_uplift': {'all': [],
                                                                           'first': {}},
                                                       'significance': 'neutral'}},
                          'is_control': False},
              'variant': {'branch_data': {'identity': {'absolute': {'all': [{'lower': 10,
                                                                             'point': 12,
                                                                             'upper': 13,
                                                                             'window_index': '1'}],
                                                                    'first': {'lower': 10,
                                                                              'point': 12,
                                                                              'upper': 13,
                                                                              'window_index': '1'}},
                                                       'difference': {'all': [],
                                                                      'first': {}},
                                                       'relative_uplift': {'all': [],
                                                                           'first': {}}},
                                          'multi-layered-homogeneous-graphic-interface_ever_used': {'absolute': {'all': [{'lower': 2,
                                                                                                                          'point': 4,
                                                                                                                          'upper': 8,
                                                                                                                          'window_index': '1'}],
                                                                                                                 'first': {'lower': 2,
                                                                                                                           'point': 4,
                                                                                                                           'upper': 8,
                                                                                                                           'window_index': '1'}},
                                                                                                    'difference': {'all': [],
                                                                                                                   'first': {}},
                                                                                                    'relative_uplift': {'all': [],
                                                                                                                        'first': {}}},
-                                         'object-based-client-driven-initiative_ever_used': {'absolute': {'all': [{'lower': 2,
-                                                                                                                   'point': 4,
-                                                                                                                   'upper': 8,
-                                                                                                                   'window_index': '1'}],
-                                                                                                          'first': {'lower': 2,
-                                                                                                                    'point': 4,
-                                                                                                                    'upper': 8,
-                                                                                                                    'window_index': '1'}},
-                                                                                             'difference': {'all': [],
-                                                                                                            'first': {}},
-                                                                                             'relative_uplift': {'all': [],
-                                                                                                                 'first': {}}},
                                          'retained': {'absolute': {'all': [],
                                                                    'first': {}},
                                                       'difference': {'all': [{'lower': -5,
                                                                               'point': -2,
                                                                               'upper': -1,
                                                                               'window_index': '1'}],
                                                                      'first': {'lower': -5,
                                                                                'point': -2,
                                                                                'upper': -1,
                                                                                'window_index': '1'}},
                                                       'relative_uplift': {'all': [],
                                                                           'first': {}},
                                                       'significance': 'negative'},
                                          'search_count': {'absolute': {'all': [],
                                                                        'first': {}},
                                                           'difference': {'all': [{'lower': 10,
                                                                                   'point': 12,
                                                                                   'upper': 13,
                                                                                   'window_index': '1'}],
                                                                          'first': {'lower': 10,
                                                                                    'point': 12,
                                                                                    'upper': 13,
                                                                                    'window_index': '1'}},
                                                           'relative_uplift': {'all': [],
                                                                               'first': {}},
                                                           'significance': 'positive'},
                                          'some_count': {'absolute': {'all': [{'lower': 10,
                                                                               'point': 12,
                                                                               'upper': 13,
                                                                               'window_index': '1'}],
                                                                      'first': {'lower': 10,
                                                                                'point': 12,
                                                                                'upper': 13,
                                                                                'window_index': '1'}},
                                                         'difference': {'all': [],
                                                                        'first': {}},
                                                         'relative_uplift': {'all': [],
                                                                             'first': {}}}},
                          'is_control': False}}}

----------------------------------------------------------------------
Ran 723 tests in 12.542s
```

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/EXP-935)
┆Issue Number: EXP-935

I just got this without using parallel:

```
======================================================================
FAIL: test_analysis_results_view_data_1_Complete (experimenter.visualization.tests.api.test_views.TestVisualizationView)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/parameterized/parameterized.py", line 533, in standalone_func
    return func(*(a + p.args), **p.kwargs)
  File "/usr/local/lib/python3.9/unittest/mock.py", line 1337, in patched
    return func(*newargs, **newkeywargs)
  File "/usr/local/lib/python3.9/site-packages/parameterized/parameterized.py", line 81, in dummy_func
    return orgfunc(*args, **kwargs)
  File "/usr/local/lib/python3.9/unittest/mock.py", line 1337, in patched
    return func(*newargs, **newkeywargs)
  File "/app/experimenter/visualization/tests/api/test_views.py", line 151, in test_analysis_results_view_data
    self.assertEqual(FULL_DATA, json_data)
AssertionError: {'daily': [{'point': 12, 'upper': 13, 'lowe[7037 chars]alse} != {'show_analysis': False, 'daily': [{'point'[5941 chars]}}}}}
  {'daily': [{'branch': 'control',
              'lower': 10,
              'metric': 'identity',
              'point': 12,
              'statistic': 'count',
              'upper': 13,
              'window_index': '1'},
             {'branch': 'variant',
              'lower': 10,
              'metric': 'identity',
              'point': 12,
              'statistic': 'count',
              'upper': 13,
              'window_index': '1'},
             {'branch': 'variant',
              'lower': 10,
              'metric': 'some_count',
              'point': 12,
              'statistic': 'mean',
              'upper': 13,
              'window_index': '1'},
             {'branch': 'variant',
              'comparison': 'difference',
              'lower': 10,
              'metric': 'search_count',
              'point': 12,
              'statistic': 'mean',
              'upper': 13,
              'window_index': '1'},
             {'branch': 'variant',
              'comparison': 'difference',
              'lower': -5,
              'metric': 'retained',
              'point': -2,
              'statistic': 'binomial',
              'upper': -1,
              'window_index': '1'},
             {'branch': 'control',
              'comparison': 'difference',
              'lower': -5,
              'metric': 'retained',
              'point': 12,
              'statistic': 'binomial',
              'upper': 13,
              'window_index': '1'},
             {'branch': 'control',
              'lower': 2,
              'metric': 'enhanced-user-facing-initiative_ever_used',
              'point': 4,
              'statistic': 'binomial',
              'upper': 8,
              'window_index': '1'},
             {'branch': 'variant',
              'lower': 2,
              'metric': 'enhanced-user-facing-initiative_ever_used',
              'point': 4,
              'statistic': 'binomial',
              'upper': 8,
              'window_index': '1'},
             {'branch': 'control',
              'lower': 2,
              'metric': 'right-sized-24hour-encryption_ever_used',
              'point': 4,
              'statistic': 'binomial',
              'upper': 8,
              'window_index': '1'},
             {'branch': 'variant',
              'lower': 2,
              'metric': 'right-sized-24hour-encryption_ever_used',
              'point': 4,
              'statistic': 'binomial',
              'upper': 8,
              'window_index': '1'}],
   'other_metrics': {'some_count': 'Some Count'},
+  'overall': {'control': {'branch_data': {'identity': {'absolute': {'all': [{'lower': 10,
-  'overall': {'control': {'branch_data': {'enhanced-user-facing-initiative_ever_used': {'absolute': {'all': [{'count': 48,
-                                                                                                              'lower': 2,
-                                                                                                              'point': 4,
-                                                                                                              'upper': 8}],
-                                                                                                     'first': {'count': 48,
-                                                                                                               'lower': 2,
-                                                                                                               'point': 4,
-                                                                                                               'upper': 8}},
-                                                                                        'difference': {'all': [],
-                                                                                                       'first': {}},
-                                                                                        'relative_uplift': {'all': [],
-                                                                                                            'first': {}}},
-                                          'identity': {'absolute': {'all': [{'lower': 10,
                                                                              'point': 12,
                                                                              'upper': 13}],
                                                                     'first': {'lower': 10,
                                                                               'point': 12,
                                                                               'upper': 13}},
                                                        'difference': {'all': [],
                                                                       'first': {}},
                                                        'percent': 50,
                                                        'relative_uplift': {'all': [],
                                                                            'first': {}}},
                                           'retained': {'absolute': {'all': [],
                                                                     'first': {}},
                                                        'difference': {'all': [{'lower': -5,
                                                                                'point': 12,
                                                                                'upper': 13},
                                                                               {'lower': -5,
                                                                                'point': 12,
                                                                                'upper': 13}],
                                                                       'first': {'lower': -5,
                                                                                 'point': 12,
                                                                                 'upper': 13}},
                                                        'relative_uplift': {'all': [],
                                                                            'first': {}},
                                                        'significance': 'neutral'},
                                           'right-sized-24hour-encryption_ever_used': {'absolute': {'all': [{'count': 48,
                                                                                                             'lower': 2,
                                                                                                             'point': 4,
                                                                                                             'upper': 8}],
                                                                                                    'first': {'count': 48,
                                                                                                              'lower': 2,
                                                                                                              'point': 4,
                                                                                                              'upper': 8}},
                                                                                       'difference': {'all': [],
                                                                                                      'first': {}},
                                                                                       'relative_uplift': {'all': [],
                                                                                                           'first': {}}}},
                           'is_control': False},
+              'variant': {'branch_data': {'identity': {'absolute': {'all': [{'lower': 10,
-              'variant': {'branch_data': {'enhanced-user-facing-initiative_ever_used': {'absolute': {'all': [{'count': 48,
-                                                                                                              'lower': 2,
-                                                                                                              'point': 4,
-                                                                                                              'upper': 8}],
-                                                                                                     'first': {'count': 48,
-                                                                                                               'lower': 2,
-                                                                                                               'point': 4,
-                                                                                                               'upper': 8}},
-                                                                                        'difference': {'all': [],
-                                                                                                       'first': {}},
-                                                                                        'relative_uplift': {'all': [],
-                                                                                                            'first': {}}},
-                                          'identity': {'absolute': {'all': [{'lower': 10,
                                                                              'point': 12,
                                                                              'upper': 13}],
                                                                     'first': {'lower': 10,
                                                                               'point': 12,
                                                                               'upper': 13}},
                                                        'difference': {'all': [],
                                                                       'first': {}},
                                                        'percent': 50,
                                                        'relative_uplift': {'all': [],
                                                                            'first': {}}},
                                           'retained': {'absolute': {'all': [],
                                                                     'first': {}},
                                                        'difference': {'all': [{'lower': -5,
                                                                                'point': -2,
                                                                                'upper': -1},
                                                                               {'lower': -5,
                                                                                'point': -2,
                                                                                'upper': -1}],
                                                                       'first': {'lower': -5,
                                                                                 'point': -2,
                                                                                 'upper': -1}},
                                                        'relative_uplift': {'all': [],
                                                                            'first': {}},
                                                        'significance': 'negative'},
                                           'right-sized-24hour-encryption_ever_used': {'absolute': {'all': [{'count': 48,
                                                                                                             'lower': 2,
                                                                                                             'point': 4,
                                                                                                             'upper': 8}],
                                                                                                    'first': {'count': 48,
                                                                                                              'lower': 2,
                                                                                                              'point': 4,
                                                                                                              'upper': 8}},
                                                                                       'difference': {'all': [],
                                                                                                      'first': {}},
                                                                                       'relative_uplift': {'all': [],
                                                                                                           'first': {}}},
                                           'search_count': {'absolute': {'all': [],
                                                                         'first': {}},
                                                            'difference': {'all': [{'lower': 10,
                                                                                    'point': 12,
                                                                                    'upper': 13}],
                                                                           'first': {'lower': 10,
                                                                                     'point': 12,
                                                                                     'upper': 13}},
                                                            'relative_uplift': {'all': [],
                                                                                'first': {}},
                                                            'significance': 'positive'},
                                           'some_count': {'absolute': {'all': [{'lower': 10,
                                                                                'point': 12,
                                                                                'upper': 13}],
                                                                       'first': {'lower': 10,
                                                                                 'point': 12,
                                                                                 'upper': 13}},
                                                          'difference': {'all': [],
                                                                         'first': {}},
                                                          'relative_uplift': {'all': [],
                                                                              'first': {}}}},
                           'is_control': False}},
   'show_analysis': False,
-  'weekly': {'control': {'branch_data': {'enhanced-user-facing-initiative_ever_used': {'absolute': {'all': [{'lower': 2,
?                                            ^^^^^^^^^^^^^^^^^^^^^  ^^^^^^^^^^^^^^^^                                   ^

+  'weekly': {'control': {'branch_data': {'identity': {'absolute': {'all': [{'lower': 10,
?                                          ++  ^  ^                                   ^^

-                                                                                                             'point': 4,
-                                                                                                             'upper': 8,
-                                                                                                             'window_index': '1'}],
-                                                                                                    'first': {'lower': 2,
-                                                                                                              'point': 4,
-                                                                                                              'upper': 8,
-                                                                                                              'window_index': '1'}},
-                                                                                       'difference': {'all': [],
-                                                                                                      'first': {}},
-                                                                                       'relative_uplift': {'all': [],
-                                                                                                           'first': {}}},
-                                         'identity': {'absolute': {'all': [{'lower': 10,
                                                                             'point': 12,
                                                                             'upper': 13,
                                                                             'window_index': '1'}],
                                                                    'first': {'lower': 10,
                                                                              'point': 12,
                                                                              'upper': 13,
                                                                              'window_index': '1'}},
                                                       'difference': {'all': [],
                                                                      'first': {}},
                                                       'relative_uplift': {'all': [],
                                                                           'first': {}}},
                                          'retained': {'absolute': {'all': [],
                                                                    'first': {}},
                                                       'difference': {'all': [{'lower': -5,
                                                                               'point': 12,
                                                                               'upper': 13,
                                                                               'window_index': '1'}],
                                                                      'first': {'lower': -5,
                                                                                'point': 12,
                                                                                'upper': 13,
                                                                                'window_index': '1'}},
                                                       'relative_uplift': {'all': [],
                                                                           'first': {}},
                                                       'significance': 'neutral'},
                                          'right-sized-24hour-encryption_ever_used': {'absolute': {'all': [{'lower': 2,
                                                                                                            'point': 4,
                                                                                                            'upper': 8,
                                                                                                            'window_index': '1'}],
                                                                                                   'first': {'lower': 2,
                                                                                                             'point': 4,
                                                                                                             'upper': 8,
                                                                                                             'window_index': '1'}},
                                                                                      'difference': {'all': [],
                                                                                                     'first': {}},
                                                                                      'relative_uplift': {'all': [],
                                                                                                          'first': {}}}},
                          'is_control': False},
-             'variant': {'branch_data': {'enhanced-user-facing-initiative_ever_used': {'absolute': {'all': [{'lower': 2,
?                                            ^^^^^^^^^^^^^^^^^^^^^  ^^^^^^^^^^^^^^^^                                   ^

+             'variant': {'branch_data': {'identity': {'absolute': {'all': [{'lower': 10,
?                                          ++  ^  ^                                   ^^

-                                                                                                             'point': 4,
-                                                                                                             'upper': 8,
-                                                                                                             'window_index': '1'}],
-                                                                                                    'first': {'lower': 2,
-                                                                                                              'point': 4,
-                                                                                                              'upper': 8,
-                                                                                                              'window_index': '1'}},
-                                                                                       'difference': {'all': [],
-                                                                                                      'first': {}},
-                                                                                       'relative_uplift': {'all': [],
-                                                                                                           'first': {}}},
-                                         'identity': {'absolute': {'all': [{'lower': 10,
                                                                             'point': 12,
                                                                             'upper': 13,
                                                                             'window_index': '1'}],
                                                                    'first': {'lower': 10,
                                                                              'point': 12,
                                                                              'upper': 13,
                                                                              'window_index': '1'}},
                                                       'difference': {'all': [],
                                                                      'first': {}},
                                                       'relative_uplift': {'all': [],
                                                                           'first': {}}},
                                          'retained': {'absolute': {'all': [],
                                                                    'first': {}},
                                                       'difference': {'all': [{'lower': -5,
                                                                               'point': -2,
                                                                               'upper': -1,
                                                                               'window_index': '1'}],
                                                                      'first': {'lower': -5,
                                                                                'point': -2,
                                                                                'upper': -1,
                                                                                'window_index': '1'}},
                                                       'relative_uplift': {'all': [],
                                                                           'first': {}},
                                                       'significance': 'negative'},
                                          'right-sized-24hour-encryption_ever_used': {'absolute': {'all': [{'lower': 2,
                                                                                                            'point': 4,
                                                                                                            'upper': 8,
                                                                                                            'window_index': '1'}],
                                                                                                   'first': {'lower': 2,
                                                                                                             'point': 4,
                                                                                                             'upper': 8,
                                                                                                             'window_index': '1'}},
                                                                                      'difference': {'all': [],
                                                                                                     'first': {}},
                                                                                      'relative_uplift': {'all': [],
                                                                                                          'first': {}}},
                                          'search_count': {'absolute': {'all': [],
                                                                        'first': {}},
                                                           'difference': {'all': [{'lower': 10,
                                                                                   'point': 12,
                                                                                   'upper': 13,
                                                                                   'window_index': '1'}],
                                                                          'first': {'lower': 10,
                                                                                    'point': 12,
                                                                                    'upper': 13,
                                                                                    'window_index': '1'}},
                                                           'relative_uplift': {'all': [],
                                                                               'first': {}},
                                                           'significance': 'positive'},
                                          'some_count': {'absolute': {'all': [{'lower': 10,
                                                                               'point': 12,
                                                                               'upper': 13,
                                                                               'window_index': '1'}],
                                                                      'first': {'lower': 10,
                                                                                'point': 12,
                                                                                'upper': 13,
                                                                                'window_index': '1'}},
                                                         'difference': {'all': [],
                                                                        'first': {}},
                                                         'relative_uplift': {'all': [],
                                                                             'first': {}}}},
                          'is_control': False}}}

----------------------------------------------------------------------
Ran 723 tests in 50.859s

FAILED (failures=1)
```

--- a/app/experimenter/visualization/tests/api/constants.py
+++ b/app/experimenter/visualization/tests/api/constants.py
@@ -2,279 +2,307 @@ from experimenter.visualization.api.v3.views import Significance
 
 
 class TestConstants:
-    DATA_IDENTITY_ROW = {
-        "point": 12,
-        "upper": 13,
-        "lower": 10,
-        "metric": "identity",
-        "statistic": "count",
-        "window_index": "1",
-    }
-    CONTROL_DATA_ROW = {
-        **DATA_IDENTITY_ROW,
-        **{
-            "branch": "control",
-        },
-    }
-    VARIANT_DATA_ROW = {
-        **DATA_IDENTITY_ROW,
-        **{
-            "branch": "variant",
-        },
-    }
-    VARIANT_DATA_DEFAULT_METRIC_ROW = {
-        **DATA_IDENTITY_ROW,
-        **{
-            "metric": "some_count",
-            "statistic": "mean",
-            "branch": "variant",
-        },
-    }
-    VARIANT_POSITIVE_SIGNIFICANCE_DATA_ROW = {
-        **DATA_IDENTITY_ROW,
-        **{
-            "comparison": "difference",
-            "metric": "search_count",
-            "statistic": "mean",
-            "branch": "variant",
-        },
-    }
-    VARIANT_NEGATIVE_SIGNIFICANCE_DATA_ROW = {
-        **VARIANT_POSITIVE_SIGNIFICANCE_DATA_ROW,
-        **{
-            "point": -2,
-            "upper": -1,
-            "lower": -5,
-            "metric": "retained",
-            "statistic": "binomial",
-        },
-    }
-    CONTROL_NEUTRAL_SIGNIFICANCE_DATA_ROW = {
-        **VARIANT_NEGATIVE_SIGNIFICANCE_DATA_ROW,
-        **{
+    @classmethod
+    def get_test_data(cls):
+        DATA_IDENTITY_ROW = {
             "point": 12,
             "upper": 13,
-            "lower": -5,
-            "branch": "control",
-        },
-    }
-    DATA_WITHOUT_POPULATION_PERCENTAGE = [
-        CONTROL_DATA_ROW,
-        VARIANT_DATA_ROW,
-        VARIANT_DATA_DEFAULT_METRIC_ROW,
-        VARIANT_POSITIVE_SIGNIFICANCE_DATA_ROW,
-        VARIANT_NEGATIVE_SIGNIFICANCE_DATA_ROW,
-        CONTROL_NEUTRAL_SIGNIFICANCE_DATA_ROW,
-    ]
+            "lower": 10,
+            "metric": "identity",
+            "statistic": "count",
+            "window_index": "1",
+        }
+        CONTROL_DATA_ROW = {
+            **DATA_IDENTITY_ROW,
+            **{
+                "branch": "control",
+            },
+        }
+        VARIANT_DATA_ROW = {
+            **DATA_IDENTITY_ROW,
+            **{
+                "branch": "variant",
+            },
+        }
+        VARIANT_DATA_DEFAULT_METRIC_ROW = {
+            **DATA_IDENTITY_ROW,
+            **{
+                "metric": "some_count",
+                "statistic": "mean",
+                "branch": "variant",
+            },
+        }
+        VARIANT_POSITIVE_SIGNIFICANCE_DATA_ROW = {
+            **DATA_IDENTITY_ROW,
+            **{
+                "comparison": "difference",
+                "metric": "search_count",
+                "statistic": "mean",
+                "branch": "variant",
+            },
+        }
+        VARIANT_NEGATIVE_SIGNIFICANCE_DATA_ROW = {
+            **VARIANT_POSITIVE_SIGNIFICANCE_DATA_ROW,
+            **{
+                "point": -2,
+                "upper": -1,
+                "lower": -5,
+                "metric": "retained",
+                "statistic": "binomial",
+            },
+        }
+        CONTROL_NEUTRAL_SIGNIFICANCE_DATA_ROW = {
+            **VARIANT_NEGATIVE_SIGNIFICANCE_DATA_ROW,
+            **{
+                "point": 12,
+                "upper": 13,
+                "lower": -5,
+                "branch": "control",
+            },
+        }
+        DATA_WITHOUT_POPULATION_PERCENTAGE = [
+            CONTROL_DATA_ROW,
+            VARIANT_DATA_ROW,
+            VARIANT_DATA_DEFAULT_METRIC_ROW,
+            VARIANT_POSITIVE_SIGNIFICANCE_DATA_ROW,
+            VARIANT_NEGATIVE_SIGNIFICANCE_DATA_ROW,
+            CONTROL_NEUTRAL_SIGNIFICANCE_DATA_ROW,
+        ]
 
-    FORMATTED_DATA_WITHOUT_POPULATION_PERCENTAGE = {
-        "control": {
-            "is_control": False,
-            "branch_data": {
-                "identity": {
-                    "absolute": {
-                        "first": {
-                            "lower": 10,
-                            "point": 12,
-                            "upper": 13,
-                            "window_index": "1",
+        FORMATTED_DATA_WITHOUT_POPULATION_PERCENTAGE = {
+            "control": {
+                "is_control": False,
+                "branch_data": {
+                    "identity": {
+                        "absolute": {
+                            "first": {
+                                "lower": 10,
+                                "point": 12,
+                                "upper": 13,
+                                "window_index": "1",
+                            },
+                            "all": [
+                                {
+                                    "lower": 10,
+                                    "point": 12,
+                                    "upper": 13,
+                                    "window_index": "1",
+                                }
+                            ],
                         },
-                        "all": [
-                            {"lower": 10, "point": 12, "upper": 13, "window_index": "1"}
-                        ],
+                        "difference": {"all": [], "first": {}},
+                        "relative_uplift": {"all": [], "first": {}},
                     },
-                    "difference": {"all": [], "first": {}},
-                    "relative_uplift": {"all": [], "first": {}},
-                },
-                "retained": {
-                    "absolute": {"all": [], "first": {}},
-                    "difference": {
-                        "all": [
-                            {
+                    "retained": {
+                        "absolute": {"all": [], "first": {}},
+                        "difference": {
+                            "all": [
+                                {
+                                    "point": 12,
+                                    "upper": 13,
+                                    "lower": -5,
+                                    "window_index": "1",
+                                }
+                            ],
+                            "first": {
                                 "point": 12,
                                 "upper": 13,
                                 "lower": -5,
                                 "window_index": "1",
-                            }
-                        ],
-                        "first": {
-                            "point": 12,
-                            "upper": 13,
-                            "lower": -5,
-                            "window_index": "1",
+                            },
                         },
+                        "relative_uplift": {"all": [], "first": {}},
+                        "significance": Significance.NEUTRAL,
                     },
-                    "relative_uplift": {"all": [], "first": {}},
-                    "significance": Significance.NEUTRAL,
                 },
             },
-        },
-        "variant": {
-            "is_control": False,
-            "branch_data": {
-                "identity": {
-                    "absolute": {
-                        "first": {
-                            "lower": 10,
-                            "point": 12,
-                            "upper": 13,
-                            "window_index": "1",
+            "variant": {
+                "is_control": False,
+                "branch_data": {
+                    "identity": {
+                        "absolute": {
+                            "first": {
+                                "lower": 10,
+                                "point": 12,
+                                "upper": 13,
+                                "window_index": "1",
+                            },
+                            "all": [
+                                {
+                                    "lower": 10,
+                                    "point": 12,
+                                    "upper": 13,
+                                    "window_index": "1",
+                                }
+                            ],
                         },
-                        "all": [
-                            {"lower": 10, "point": 12, "upper": 13, "window_index": "1"}
-                        ],
+                        "difference": {"all": [], "first": {}},
+                        "relative_uplift": {"all": [], "first": {}},
                     },
-                    "difference": {"all": [], "first": {}},
-                    "relative_uplift": {"all": [], "first": {}},
-                },
-                "search_count": {
-                    "absolute": {"all": [], "first": {}},
-                    "difference": {
-                        "first": {
-                            "lower": 10,
-                            "point": 12,
-                            "upper": 13,
-                            "window_index": "1",
+                    "search_count": {
+                        "absolute": {"all": [], "first": {}},
+                        "difference": {
+                            "first": {
+                                "lower": 10,
+                                "point": 12,
+                                "upper": 13,
+                                "window_index": "1",
+                            },
+                            "all": [
+                                {
+                                    "lower": 10,
+                                    "point": 12,
+                                    "upper": 13,
+                                    "window_index": "1",
+                                }
+                            ],
                         },
-                        "all": [
-                            {"lower": 10, "point": 12, "upper": 13, "window_index": "1"}
-                        ],
+                        "relative_uplift": {"all": [], "first": {}},
+                        "significance": Significance.POSITIVE,
                     },
-                    "relative_uplift": {"all": [], "first": {}},
-                    "significance": Significance.POSITIVE,
-                },
-                "some_count": {
-                    "absolute": {
-                        "first": {
-                            "lower": 10,
-                            "point": 12,
-                            "upper": 13,
-                            "window_index": "1",
+                    "some_count": {
+                        "absolute": {
+                            "first": {
+                                "lower": 10,
+                                "point": 12,
+                                "upper": 13,
+                                "window_index": "1",
+                            },
+                            "all": [
+                                {
+                                    "lower": 10,
+                                    "point": 12,
+                                    "upper": 13,
+                                    "window_index": "1",
+                                }
+                            ],
                         },
-                        "all": [
-                            {"lower": 10, "point": 12, "upper": 13, "window_index": "1"}
-                        ],
+                        "difference": {"all": [], "first": {}},
+                        "relative_uplift": {"all": [], "first": {}},
                     },
-                    "difference": {"all": [], "first": {}},
-                    "relative_uplift": {"all": [], "first": {}},
-                },
-                "retained": {
-                    "absolute": {"all": [], "first": {}},
-                    "difference": {
-                        "first": {
-                            "point": -2,
-                            "upper": -1,
-                            "lower": -5,
-                            "window_index": "1",
-                        },
-                        "all": [
-                            {
+                    "retained": {
+                        "absolute": {"all": [], "first": {}},
+                        "difference": {
+                            "first": {
                                 "point": -2,
                                 "upper": -1,
                                 "lower": -5,
                                 "window_index": "1",
-                            }
-                        ],
+                            },
+                            "all": [
+                                {
+                                    "point": -2,
+                                    "upper": -1,
+                                    "lower": -5,
+                                    "window_index": "1",
+                                }
+                            ],
+                        },
+                        "relative_uplift": {"all": [], "first": {}},
+                        "significance": Significance.NEGATIVE,
                     },
-                    "relative_uplift": {"all": [], "first": {}},
-                    "significance": Significance.NEGATIVE,
                 },
             },
-        },
-    }
+        }
 
-    FORMATTED_DATA_WITH_POPULATION_PERCENTAGE = {
-        "control": {
-            "is_control": False,
-            "branch_data": {
-                "identity": {
-                    "absolute": {
-                        "first": {"lower": 10, "point": 12, "upper": 13},
-                        "all": [{"lower": 10, "point": 12, "upper": 13}],
+        FORMATTED_DATA_WITH_POPULATION_PERCENTAGE = {
+            "control": {
+                "is_control": False,
+                "branch_data": {
+                    "identity": {
+                        "absolute": {
+                            "first": {"lower": 10, "point": 12, "upper": 13},
+                            "all": [{"lower": 10, "point": 12, "upper": 13}],
+                        },
+                        "difference": {"all": [], "first": {}},
+                        "relative_uplift": {"all": [], "first": {}},
+                        "percent": 50,
                     },
-                    "difference": {"all": [], "first": {}},
-                    "relative_uplift": {"all": [], "first": {}},
-                    "percent": 50,
-                },
-                "retained": {
-                    "absolute": {"all": [], "first": {}},
-                    "difference": {
-                        "all": [
-                            {
+                    "retained": {
+                        "absolute": {"all": [], "first": {}},
+                        "difference": {
+                            "all": [
+                                {
+                                    "point": 12,
+                                    "upper": 13,
+                                    "lower": -5,
+                                },
+                                {
+                                    "point": 12,
+                                    "upper": 13,
+                                    "lower": -5,
+                                },
+                            ],
+                            "first": {
                                 "point": 12,
                                 "upper": 13,
                                 "lower": -5,
                             },
-                            {
-                                "point": 12,
-                                "upper": 13,
-                                "lower": -5,
-                            },
-                        ],
-                        "first": {
-                            "point": 12,
-                            "upper": 13,
-                            "lower": -5,
                         },
+                        "relative_uplift": {"all": [], "first": {}},
+                        "significance": Significance.NEUTRAL,
                     },
-                    "relative_uplift": {"all": [], "first": {}},
-                    "significance": Significance.NEUTRAL,
                 },
             },
-        },
-        "variant": {
-            "is_control": False,
-            "branch_data": {
-                "identity": {
-                    "absolute": {
-                        "first": {"lower": 10, "point": 12, "upper": 13},
-                        "all": [{"lower": 10, "point": 12, "upper": 13}],
-                    },
-                    "difference": {"all": [], "first": {}},
-                    "relative_uplift": {"all": [], "first": {}},
-                    "percent": 50,
-                },
-                "search_count": {
-                    "absolute": {"all": [], "first": {}},
-                    "difference": {
-                        "first": {"lower": 10, "point": 12, "upper": 13},
-                        "all": [{"lower": 10, "point": 12, "upper": 13}],
-                    },
-                    "relative_uplift": {"all": [], "first": {}},
-                    "significance": Significance.POSITIVE,
-                },
-                "some_count": {
-                    "absolute": {
-                        "first": {"lower": 10, "point": 12, "upper": 13},
-                        "all": [{"lower": 10, "point": 12, "upper": 13}],
-                    },
-                    "difference": {"all": [], "first": {}},
-                    "relative_uplift": {"all": [], "first": {}},
-                },
-                "retained": {
-                    "absolute": {"all": [], "first": {}},
-                    "difference": {
-                        "first": {
-                            "point": -2,
-                            "upper": -1,
-                            "lower": -5,
+            "variant": {
+                "is_control": False,
+                "branch_data": {
+                    "identity": {
+                        "absolute": {
+                            "first": {"lower": 10, "point": 12, "upper": 13},
+                            "all": [{"lower": 10, "point": 12, "upper": 13}],
                         },
-                        "all": [
-                            {
+                        "difference": {"all": [], "first": {}},
+                        "relative_uplift": {"all": [], "first": {}},
+                        "percent": 50,
+                    },
+                    "search_count": {
+                        "absolute": {"all": [], "first": {}},
+                        "difference": {
+                            "first": {"lower": 10, "point": 12, "upper": 13},
+                            "all": [{"lower": 10, "point": 12, "upper": 13}],
+                        },
+                        "relative_uplift": {"all": [], "first": {}},
+                        "significance": Significance.POSITIVE,
+                    },
+                    "some_count": {
+                        "absolute": {
+                            "first": {"lower": 10, "point": 12, "upper": 13},
+                            "all": [{"lower": 10, "point": 12, "upper": 13}],
+                        },
+                        "difference": {"all": [], "first": {}},
+                        "relative_uplift": {"all": [], "first": {}},
+                    },
+                    "retained": {
+                        "absolute": {"all": [], "first": {}},
+                        "difference": {
+                            "first": {
                                 "point": -2,
                                 "upper": -1,
                                 "lower": -5,
                             },
-                            {
-                                "point": -2,
-                                "upper": -1,
-                                "lower": -5,
-                            },
-                        ],
+                            "all": [
+                                {
+                                    "point": -2,
+                                    "upper": -1,
+                                    "lower": -5,
+                                },
+                                {
+                                    "point": -2,
+                                    "upper": -1,
+                                    "lower": -5,
+                                },
+                            ],
+                        },
+                        "relative_uplift": {"all": [], "first": {}},
+                        "significance": Significance.NEGATIVE,
                     },
-                    "relative_uplift": {"all": [], "first": {}},
-                    "significance": Significance.NEGATIVE,
                 },
             },
-        },
-    }
+        }
+
+        return (
+            DATA_WITHOUT_POPULATION_PERCENTAGE,
+            FORMATTED_DATA_WITHOUT_POPULATION_PERCENTAGE,
+            FORMATTED_DATA_WITH_POPULATION_PERCENTAGE,
+        )

--- a/app/experimenter/visualization/tests/api/test_views.py
+++ b/app/experimenter/visualization/tests/api/test_views.py
@@ -16,6 +16,8 @@ from experimenter.visualization.tests.api.constants import TestConstants
 
 @override_settings(FEATURE_ANALYSIS=False)
 class TestVisualizationView(TestCase):
+    maxDiff = None
+
     @parameterized.expand(
         [
             NimbusExperiment.Status.ACCEPTED,
@@ -102,24 +104,33 @@ class TestVisualizationView(TestCase):
             )
 
     @parameterized.expand(
-        [NimbusExperiment.Status.ACCEPTED, NimbusExperiment.Status.COMPLETE]
+        [
+            NimbusExperiment.Status.COMPLETE,
+            NimbusExperiment.Status.ACCEPTED,
+        ]
     )
     @patch("django.core.files.storage.default_storage.open")
     @patch("django.core.files.storage.default_storage.exists")
     def test_analysis_results_view_data(self, status, mock_exists, mock_open):
         user_email = "user@example.com"
 
+        (
+            DATA_WITHOUT_POPULATION_PERCENTAGE,
+            FORMATTED_DATA_WITHOUT_POPULATION_PERCENTAGE,
+            FORMATTED_DATA_WITH_POPULATION_PERCENTAGE,
+        ) = TestConstants.get_test_data()
+
         FULL_DATA = {
-            "daily": TestConstants.DATA_WITHOUT_POPULATION_PERCENTAGE,
-            "weekly": TestConstants.FORMATTED_DATA_WITHOUT_POPULATION_PERCENTAGE,
-            "overall": TestConstants.FORMATTED_DATA_WITH_POPULATION_PERCENTAGE,
+            "daily": DATA_WITHOUT_POPULATION_PERCENTAGE,
+            "weekly": FORMATTED_DATA_WITHOUT_POPULATION_PERCENTAGE,
+            "overall": FORMATTED_DATA_WITH_POPULATION_PERCENTAGE,
             "other_metrics": {"some_count": "Some Count"},
             "show_analysis": False,
         }
 
         class File:
             def read(self):
-                return json.dumps(TestConstants.DATA_WITHOUT_POPULATION_PERCENTAGE)
+                return json.dumps(DATA_WITHOUT_POPULATION_PERCENTAGE)
 
         mock_open.return_value = File()
         mock_exists.return_value = True
@@ -127,15 +138,16 @@ class TestVisualizationView(TestCase):
         experiment = NimbusExperimentFactory.create_with_status(
             target_status=status, probe_sets=[primary_probe_set]
         )
+
         experiment.probe_sets.add(
             NimbusProbeSetFactory.create(),
             through_defaults={"is_primary": False},
         )
 
         self.add_all_probe_set_data(
-            TestConstants.DATA_WITHOUT_POPULATION_PERCENTAGE,
-            TestConstants.FORMATTED_DATA_WITH_POPULATION_PERCENTAGE,
-            TestConstants.FORMATTED_DATA_WITHOUT_POPULATION_PERCENTAGE,
+            DATA_WITHOUT_POPULATION_PERCENTAGE,
+            FORMATTED_DATA_WITH_POPULATION_PERCENTAGE,
+            FORMATTED_DATA_WITHOUT_POPULATION_PERCENTAGE,
             experiment.primary_probe_sets,
         )
 

